compute/metadata/metadata.go

```diff
@@ -381,19 +523,23 @@ func (c *Client) Get(suffix string) (string, error) {
 //
 // If the requested metadata is not defined, the returned error will
 // be of type NotDefinedError.
-func (c *Client) GetWithContext(ctx context.Context, suffix string) (string, error) {
+//
+// NOTE: Without an extra deadline in the context this call can take in the
+// worst case, with internal backoff retries, up to 15 seconds (e.g. when server
+// is responding slowly). Pass context with additional timeouts when needed.
+func (c *client) GetWithContext(ctx context.Context, suffix string) (string, error) {
```

Comment:
oops missed the embedding, I see. Any reason for this indirection with `client`. I don't think I see a reason why not to just put everything on `Client`. Also, I believe the way it is embedded now makes the method semantics on `client` actually be non-pointer if I am reading correctly.

Reply:
Happy to revert, my reasoning was to make sure we use type safety to ensure defaultClient only have context methods. Can see how it could be confusing. 

Let's revert, we can add later if we see we actually make mistakes later on without it 🤗